src/components/login-required.tsx

```diff
@@ -14,7 +14,7 @@ const LoginRequired: FunctionComponent<LoginRequiredParams> = ({
     if (loginRequired) {
       router.push('/login')
     }
-  }, [])
+  }, [loginRequired])
```

Comment:
this is a fairly common and difficult issue

when the boolean changes, we want them to login, so it should re-run